js/src/wp-seo-replacevar-plugin.js

```diff
@@ -127,6 +130,60 @@ import "./helpers/babel-polyfill";
 		}
 	};
 
+	/**
+	 * Subscribes to Gutenberg to watch a possible parent page change.
+	 *
+	 * @return {void}
+	 */
+	YoastReplaceVarPlugin.prototype.subscribeToGutenberg = function() {
+		if ( ! isGutenbergDataAvailable() ) {
+			return;
+		}
+
+		let fetchedParents = { 0: "" };
+		let currentParent  = null;
+		const wpData       = wp.data;
+		wpData.subscribe(
+			function() {
+				let newParent = wpData.select( "core/editor" ).getEditedPostAttribute( "parent" );
+
+				if ( typeof newParent === "undefined" || currentParent === newParent ) {
+					return;
+				}
+
+				currentParent = newParent;
+
+				if ( newParent < 1 ) {
+					this._currentParentPageTitle = "";
+
+					this.declareReloaded();
+
+					return;
+				}
+
+				if ( ! isUndefined( fetchedParents[ newParent ] ) ) {
+					this._currentParentPageTitle = fetchedParents[ newParent ];
+
+					return;
+				}
+
+				jQuery.ajax(
```

Comment:
I would prefer it if you used the wordpress javascript client: http://v2.wp-api.org/extending/javascript-client/

```
const page = new wp.api.models.Page( { id: parentId } );
page.fetch().then( callback );
```

Reply:
This also removes the need to define the nonce manually.

Reply:
No